sdk/log/provider.go

```diff
@@ -196,7 +197,11 @@ func (fn loggerProviderOptionFunc) apply(c providerConfig) providerConfig {
 // go.opentelemetry.io/otel/sdk/resource package will be used.
 func WithResource(res *resource.Resource) LoggerProviderOption {
 	return loggerProviderOptionFunc(func(cfg providerConfig) providerConfig {
-		cfg.resource = res
+		var err error
+		cfg.resource, err = resource.Merge(resource.Environment(), res)
```

Comment:
for all signals: should it not merge the passed resource with `resource.Default()` so that `defaultServiceNameDetector` and `telemetrySDK` are also applied?

Reply:
That's a good question and I asked myself the same. I think it would be the behavior that I would expect intuitively. However, my understanding is that the spec says something else:

https://opentelemetry.io/docs/specs/otel/resource/sdk/#sdk-provided-resource-attributes

> The SDK MUST provide access to a Resource with at least the attributes listed at [Semantic Attributes with SDK-provided Default Value](https://opentelemetry.io/docs/specs/semconv/resource/#semantic-attributes-with-sdk-provided-default-value). This resource MUST be associated with a TracerProvider or MeterProvider if another resource was not explicitly specified.
>
> Note: This means that it is possible to create and associate a resource that does not have all or any of the SDK-provided attributes present. However, that does not happen by default. If a user wants to combine custom attributes with the default resource, they can use [Merge](https://opentelemetry.io/docs/specs/otel/resource/sdk/#merge) with their custom resource or specify their attributes by implementing [Custom resource detectors](https://opentelemetry.io/docs/specs/otel/resource/sdk/#detecting-resource-information-from-the-environment) instead of explicitly associating a resource.

I read this as: If users provide resource information explicitly, the SDK discards should default resources attributes. If users want them to not be discarded, they would need to merge these defaults with the additional resource attributes they want. I might totally be reading this wrong though, so I'm looking forward to what other people think about this.

The section about how to treat [OTEL_RESOURCE_ATTRIBUTES](https://opentelemetry.io/docs/specs/otel/resource/sdk/#specifying-resource-information-via-an-environment-variable) only explicitly talks about merging whatever OTEL_RESOURCE_ATTRIBUTES provides with user-provided resource attributes (read: in-code configuration), not about merging with the default resource.

Reply:
Another data point: I thought it might be worth checking out what other SDKs are doing, and at least the Node.js SDK behaves in the way @pellared described, that is, if I run the basic Node.js SDK example from [here](https://opentelemetry.io/docs/languages/js/getting-started/nodejs/) like this:

```
OTEL_RESOURCE_ATTRIBUTES="service.namespace=my-service-namesspace,service.name=my-service,service.version=1.2.3" node --require ./instrumentation.js app.js`
```

...I see  that env vars and default resource attributes are merged:

```
resource: {
    attributes: {
      'service.name': 'my-service',
      'telemetry.sdk.language': 'nodejs',
      'telemetry.sdk.name': 'opentelemetry',
      'telemetry.sdk.version': '1.26.0',
      'service.namespace': 'my-service-namesspace',
      'service.version': '1.2.3',
      'process.pid': 85924,
      'process.executable.name': 'node',
      ...
```

Reply:
@basti1302, nice research.

Based on the spec I would say that we should probably merge with `defaultServiceNameDetector` and `fromEnv` dectectors (without `telemetrySDK` detector). 

For know, what this PR is proposing is probably good enough. The question is if we should do any follow-up?

Reply:
Yes on both accounts: 

(1) I would leave the question about the default resource out of scope for this PR. From this end, I think this could be merged as is.

(2) For the wider perspective, I think a good goal would be to have all SDKs behave consistently, and in line with what the spec says. Maybe the spec could also be a bit clearer on this point. I am not exactly sure what the best next steps are for that. Maybe this should be brought up with one of the SIGs first? (probably the general spec sig?) As far as I can tell, neither Go nor Node.js currently do what the spec says, so apparently there is some room for confusion.